sqlx/sqlx.go

```diff
@@ -0,0 +1,86 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pkg/errors"
+)
+
+type Client struct {
+	*sqlx.DB
```

Comment:
did we consider doing `goqu` directly? i see it being used more and more for query building. for using `ScanX` features in goqu, we need to wrap the `*sql.DB` into `*goqu.Database` using `goqu.New()`..

if we start using this `Client`, we will not be able to that easily i think..

Reply:
I think for query builder, we should use it just for `building sql query`. Scanning result rows could be done with other tools or even manually.

Reply:
Why add more dependencies when it provides what we need ? Scanning manually can still be done.

Have you seen any issues with goqu scan related functionalities?

For us, using sqlx or goqu is no different since both are just external dependencies.

Reply:
oh. if we are going to just use the query builder part, [squirrel](https://github.com/Masterminds/squirrel) feels much better (bigger community, more frequently looked after, used by a lot of projects).. anyways, that's a separate discussion. but just mentioning here.

Reply:
+1 for that. I also personally prefer `squirrel` rather than goqu.

Reply:
So, what are our thoughts on this? Shall we leave choice of SQL query builder outside of salt? Where we can recommend to use `squirrel` but would still let folks use `goqu` or  others given project requires it.

Reply:
Yes I think we could leave that choice for each service for now.

Reply:
> Where we can recommend to use squirrel but would still let folks use goqu or others given project requires it.

Should we leave it out for now? sometime I even skip using query builder and use a simple `SELECT fieldA FROM table WHERE id = $1` for simple thing.

IMHO querybuilder (squirell / goku) is just a refactoring process because query sometimes can get messy and hard to read especially for complex and dynamic query.